asn1/CommonLedgerProtocol.asn

```diff
@@ -0,0 +1,107 @@
+CommonLedgerProtocol
+DEFINITIONS
+AUTOMATIC TAGS ::=
+BEGIN
+
+IMPORTS
+    UInt8,
+    UInt32,
+    UInt64,
+    UInt128,
+    UInt256
+    FROM GenericTypes
+
+    Address,
+    Timestamp
+    FROM InterledgerTypes
+
+    InterledgerPacket
+    FROM InterledgerPacket
+;
+
+SideProtocolData ::= SEQUENCE OF SEQUENCE {
+  protocolName IA5String,
+  protocolData OCTET STRING
+}
+
+-- InterledgerPrepare, InterledgerFulfill, InterlegerReject, InterledgerRequest, and CustomRequest are the request types.
+-- Each request should have a unique requestId..
+
+InterledgerPrepare ::= SEQUENCE {
+  transferId UInt128,
+  amount UInt64,
+  executionCondition UInt256,
+  expiresAt Timestamp,
+  packet InterledgerPacket,
+  -- Additional data for protocol extensibility
+  sideProtocolData SideProtocolData
+}
+
+InterledgerFulfill ::= SEQUENCE {
+  transferId UInt128,
+  fulfillment UInt256,
+  -- Additional data for protocol extensibility
+  sideProtocolData SideProtocolData
+}
+
+InterledgerReject ::= SEQUENCE {
+  transferId UInt128,
+  rejectionReason InterledgerPacket,
+  -- Additional data for protocol extensibility
+  sideProtocolData SideProtocolData
+}
+
+InterledgerRequest ::= SEQUENCE {
+  packet InterledgerPacket,
+  -- Additional data for protocol extensibility
+  sideProtocolData SideProtocolData
+}
+
+CustomRequest ::= SideProtocolData
+
+-- Ack, InterledgerResponse, InterledgerError, and LedgerLevelError are the response types.
+-- When using these in a CLP packet, the requestId should match
+-- the requestId of the request they respond to.
+
+Ack ::= SideProtocolData
```

Comment:
I'd prefer if an Ack was simple `Ack ::= NULL`.

If there is a need to return a response that carries side protocol data then define a `LocalResponse :: = SideProtocolData`